pkg/p2p/libp2p/internal/handshake/handshake_test.go

```diff
@@ -382,15 +520,80 @@ func TestHandle(t *testing.T) {
 			t.Fatal(err)
 		}
 
-		testInfo(t, nodeInfo, Info{
-			Address:   swarm.NewAddress(got.Syn.Address),
+		testInfo(t, node1Info, Info{
+			Overlay:   swarm.NewAddress(got.Syn.BzzAddress.Overlay),
+			Underlay:  got.Syn.BzzAddress.Underlay,
 			NetworkID: got.Syn.NetworkID,
 			Light:     got.Syn.Light,
 		})
 
-		_, err = handshakeService.Handle(stream1, info.ID)
+		_, err = handshakeService.Handle(stream1, node2AddrInfo.ID)
 		if err != ErrHandshakeDuplicate {
-			t.Fatalf("expected %s err, got %s err", ErrHandshakeDuplicate, err)
+			t.Fatalf("expected %s, got %s", ErrHandshakeDuplicate, err)
+		}
+	})
+
+	t.Run("Error - invalid ack", func(t *testing.T) {
+		handshakeService, err := New(node1Info.Overlay, string(node1Info.Underlay), signer1, 0, logger)
+		if err != nil {
+			t.Fatal(err)
+		}
+
+		var buffer1 bytes.Buffer
+		var buffer2 bytes.Buffer
+		stream1 := mock.NewStream(&buffer1, &buffer2)
+		stream2 := mock.NewStream(&buffer2, &buffer1)
+
+		w, _ := protobuf.NewWriterAndReader(stream2)
+		if err := w.WriteMsg(&pb.Syn{
+			BzzAddress: node2BzzAddress,
+			NetworkID:  node2Info.NetworkID,
+			Light:      node2Info.Light,
+		}); err != nil {
+			t.Fatal(err)
+		}
+
+		if err := w.WriteMsg(&pb.Ack{BzzAddress: node2BzzAddress}); err != nil {
+			t.Fatal(err)
+		}
+
+		_, err = handshakeService.Handle(stream1, node2AddrInfo.ID)
+		if err != ErrInvalidAck {
+			t.Fatalf("expected %s, got %s", ErrInvalidAck, err)
+		}
+	})
+
+	t.Run("ERROR - invalid signature ", func(t *testing.T) {
+		handshakeService, err := New(node1Info.Overlay, string(node1Info.Underlay), signer1, 0, logger)
+		if err != nil {
+			t.Fatal(err)
+		}
+
+		var buffer1 bytes.Buffer
+		var buffer2 bytes.Buffer
+		stream1 := mock.NewStream(&buffer1, &buffer2)
+		stream2 := mock.NewStream(&buffer2, &buffer1)
+
+		w, _ := protobuf.NewWriterAndReader(stream2)
+		if err := w.WriteMsg(&pb.Syn{
+			BzzAddress: &pb.BzzAddress{
+				Underlay:  node2BzzAddress.Underlay,
+				Signature: []byte("wrong signature"),
+				Overlay:   node2BzzAddress.Overlay,
+			},
+			NetworkID: node2Info.NetworkID,
+			Light:     node2Info.Light,
+		}); err != nil {
+			t.Fatal(err)
+		}
+
+		res, err := handshakeService.Handle(stream1, node2AddrInfo.ID)
+		if res != nil {
+			t.Fatal("res should be nil")
+		}
+
+		if err != ErrInvalidSignature {
```

Comment:
this made me noticed that this test is not namespaced under `handshake_test` but just `handshake`. I guess we'd like to change that @janos?

Reply:
Fixed :)